main.tsp

```diff
@@ -1,278 +0,0 @@
-  labels: TodoLabel[];
-}
-
-union TodoLabel {
-  string,
-  string[],
-  TodoLabelRecord,
-  TodoLabelRecord[],
-}
-
-model TodoLabelRecord {
-  name: string;
-
-  @format("^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$")
-  color?: string;
-}
-
-union TodoAttachment {
-  file: TodoFileAttachment,
-  url: TodoUrlAttachment,
-}
-
-model TodoUrlAttachment {
-  /** A description of the URL */
-  description: string;
-
-  /** The url */
-  url: url;
-}
-
-model TodoFileAttachment {
-  /** The todo item this is attached to */
-  todoItemId: TodoItem.id;
-
-  /** The file name of the attachment */
-  @maxLength(255)
-  filename: string;
-
-  /** The media type of the attachment */
-  mediaType: string;
-
-  /** The url where the attachment can be downloaded from */
-  url: url;
-}
-
-enum TodoItemStatus {
-  NotStarted,
-  InProgress,
-  Completed,
-}
-
-@error
-model Error {
-  /** A machine readable error code */
-  code: string;
-
-  /** A human readable message */
-  message: string;
-}
-
-namespace Users {
-  // would prefer to extend
-  // https://github.com/microsoft/typespec/issues/2922
-
-  model UserCreatedResponse {
-    ...User;
-    ...OkResponse;
-
-    /** The token to use to construct the validate email address url */
-    token: string;
-  }
-
-  /** The user already exists */
-  model UserExistsResponse extends Error {
-    ...ConflictResponse;
-  }
-
-  /** The user is invalid (e.g. forgot to enter email address) */
-  model InvalidUserResponse extends Error {
-    @statusCode statusCode: 422;
-  }
-
-  @route("/users")
-  @post
-  op create(
-    user: User,
-  ): UserCreatedResponse | UserExistsResponse | InvalidUserResponse;
-
-  @route("validate")
-  @get
-  op validate(@query token: string): OkResponse | InvalidUserResponse;
-
-  @route("login")
-  @post
-  op login(
-    username: User.username,
-    password: User.password,
-  ): OkResponse | UnauthorizedResponse;
-
-  @route("logout")
-  @get
-  op logout(): OkResponse;
-
-  /** Sends a reset token to the user's email address */
-  @route("forgot-password")
-  @post
-  op forgotPassword(email: User.email): OkResponse | NotFoundResponse;
-
-  @route("reset-password")
-  @get
-  op resetPassword(@query resetToken: string): OkResponse | NotFoundResponse;
-}
-
-@route("items")
-namespace TodoItems {
-  model PaginationControls {
-    /** The limit to the number of items */
-    @query limit: int32 = 50;
-
-    /** The offset to start paginating at */
-    @query offset: int32 = 0;
-  }
-
-  model TodoPage {
-    /** The items in the page */
-    items: TodoItem[];
-
-    /** The number of items returned in this page */
-    pageSize: int32;
-
-    /** The total number of items */
-    totalSize: int32;
-
-    ...PaginationControls;
-
-    /** A link to the previous page, if it exists */
-    prevLink?: url;
-
-    /** A link to the next page, if it exists */
-    nextLink?: url;
-  }
-
-  // deeply annoying that I have to copy/paste this...
```

Comment:
If any of these properties become optional (which I think I suggested), they also have be made nullable. An emitter could/can use visibility/mutability + content-type to infer this model.

Reply:
I don't think this is the case - the OpenAPI emitter anyway doesn't do patchification.

Reply:
My question is if the openapi emitter **should** do the content-type:ification. OpenAPI has (or at least to have) a weird hint-hint, nudge-nudge, you-know-what-I meant interaction between model definitions/schemas and content-types other than json (with the only formal "bridge" being to XML). In OpenAPI2, nullable wasn't a thing, so you couldn't describe json merge patch without resorting to (admittedly pretty dang well known) vendor extensions.